packages/fxa-graphql-api/src/lib/resolvers/account-resolver.ts

```diff
@@ -1,31 +1,142 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
+import getStream from 'get-stream';
 import { GraphQLResolveInfo } from 'graphql';
 import {
   parseResolveInfo,
   simplifyParsedResolveInfoFragmentWithType,
 } from 'graphql-parse-resolve-info';
-import { Ctx, Query, Resolver, FieldResolver, Root, Info } from 'type-graphql';
+import {
+  Arg,
+  Ctx,
+  FieldResolver,
+  Info,
+  Mutation,
+  Query,
+  Resolver,
+  Root,
+} from 'type-graphql';
 
-import { accountByUid, Account, AccountOptions } from '../db/models/auth';
+import { Account, accountByUid, AccountOptions } from '../db/models/auth';
 import { profileByUid, selectedAvatar } from '../db/models/profile';
+import { CatchGatewayError } from '../error';
 import { Context } from '../server';
 import { Account as AccountType } from './types/account';
+import {
+  EmailInput,
+  UpdateAvatarInput,
+  UpdateDisplayNameInput,
+} from './types/input';
+import {
+  BasicPayload,
+  UpdateAvatarPayload,
+  UpdateDisplayNamePayload,
+} from './types/payload';
 
-@Resolver(of => AccountType)
+@Resolver((of) => AccountType)
 export class AccountResolver {
-  @Query(returns => AccountType, { nullable: true })
+  @Mutation((returns) => UpdateDisplayNamePayload, {
+    description: 'Update the display name.',
+  })
+  @CatchGatewayError
+  public async updateDisplayName(
+    @Ctx() context: Context,
+    @Arg('input', (type) => UpdateDisplayNameInput)
+    input: UpdateDisplayNameInput
+  ): Promise<UpdateDisplayNamePayload> {
+    const result = await context.dataSources.profileAPI.updateDisplayName(
+      input.displayName
+    );
+    return {
+      clientMutationId: input.clientMutationId,
+      displayName: result ? input.displayName : undefined,
+    };
+  }
+
+  @Mutation((returns) => UpdateAvatarPayload, {
+    description: 'Update the avatar in use.',
+  })
+  @CatchGatewayError
+  public async updateAvatar(
+    @Ctx() context: Context,
+    @Arg('input', (type) => UpdateAvatarInput) input: UpdateAvatarInput
+  ): Promise<UpdateAvatarPayload> {
+    const file = await input.file;
+    const fileData = await getStream.buffer(file.createReadStream());
```

Comment:
Looks like this loads the incoming image into memory before sending via superagent in `avatarUpload`. 

[Could we pipe the `input.file` stream to the profile-server POST request instead?](https://visionmedia.github.io/superagent/#piping-data) Maybe not worth thinking too hard about since we're hopefully talking about a couple megabytes here.

Reply:
Yes, we could pipe it into the request, the actual restriction in profile server on uploads is 1MB. nginx will typically buffer a large portion of that both in send and receive buffers which are in front of our apps, so I wasn't too worried about it here. It's worth remembering if we see any funky behavior though.